mllib/src/main/scala/org/apache/spark/mllib/stat/test/ADTest.scala

```diff
@@ -0,0 +1,295 @@
+    private val theoretical = new ExponentialDistribution(params(0))
+
+    private val rawCVs = ListMap(
+      0.15 -> 0.922, 0.10 -> 1.078,
+      0.05 -> 1.341, 0.025 -> 1.606, 0.01 -> 1.957
+    )
+
+    def cdf(x: Double): Double = theoretical.cumulativeProbability(x)
+
+    def getCVs(n: Double): Map[Double, Double] = {
+      rawCVs.map { case (sig, cv) => sig -> cv / (1 + 0.6 / n)}
+    }
+  }
+
+  // Normal Distribution
+  class ADNormal(val params: Array[Double]) extends ADTheoreticalDist {
+    private val theoretical = new NormalDistribution(params(0), params(1))
+
+    private val rawCVs = ListMap(
+      0.15 -> 0.576, 0.10 -> 0.656,
+      0.05 -> 0.787, 0.025 -> 0.918, 0.01 -> 1.092
+    )
+
+    def cdf(x: Double): Double = theoretical.cumulativeProbability(x)
+
+    def getCVs(n: Double): Map[Double, Double] = {
+      rawCVs.map { case (sig, cv) => sig -> cv / (1 + 4.0 / n - 25.0 / (n * n)) }
+    }
+  }
+
+  // Gumbel distribution
+  class ADGumbel(val params: Array[Double]) extends ADTheoreticalDist {
+    private val theoretical = new GumbelDistribution(params(0), params(1))
+
+    private val rawCVs = ListMap(
+      0.25 -> 0.474, 0.10 -> 0.637,
+      0.05 -> 0.757, 0.025 -> 0.877, 0.01 -> 1.038
+    )
+
+    def cdf(x: Double): Double = theoretical.cumulativeProbability(x)
+
+    def getCVs(n: Double): Map[Double, Double] = {
+      rawCVs.map { case (sig, cv) => sig -> cv / (1 + 0.2 / math.sqrt(n))}
+    }
+  }
+
+  // Logistic distribution
+  class ADLogistic(val params: Array[Double]) extends ADTheoreticalDist {
+    private val theoretical = new LogisticDistribution(params(0), params(1))
+
+    private val rawCVs = ListMap(
+      0.25 -> 0.426, 0.10 -> 0.563, 0.05 -> 0.660,
+      0.025 -> 0.769, 0.01 -> 0.906, 0.005 -> 1.010
+    )
+
+    def cdf(x: Double): Double = theoretical.cumulativeProbability(x)
+
+    def getCVs(n: Double): Map[Double, Double] = {
+      rawCVs.map { case (sig, cv) => sig -> cv / (1 + 0.25 / n)}
+    }
+  }
+
+  // Weibull distribution
+  class ADWeibull(val params: Array[Double]) extends ADTheoreticalDist {
+    private val theoretical = new WeibullDistribution(params(0), params(1))
+
+    private val rawCVs = ListMap(
+      0.25 -> 0.474, 0.10 -> 0.637,
+      0.05 -> 0.757, 0.025 -> 0.877, 0.01 -> 1.038
+    )
+
+    def cdf(x: Double): Double = theoretical.cumulativeProbability(x)
+
+    def getCVs(n: Double): Map[Double, Double] = {
+      rawCVs.map { case (sig, cv) => sig -> cv / (1 + 0.2 / math.sqrt(n))}
+    }
+  }
+
+  /**
+   * Perform a one sample Anderson Darling test
+   * @param data data to test for a given distribution
+   * @param distName name of theoretical distribution: currently supports standard normal,
+   *            exponential, gumbel, logistic, weibull
+   * @param params optional variable-length argument providing parameters for given distribution,
+   *               otherwise they are estimated from sample but in both cases we adjust critical
```

Comment:
sorry no, old comment. That used to be the case. No longer. I double checked comments in the latest, so this issues should be solved. Thanks